p2p/background/router.go

```diff
@@ -47,96 +66,103 @@ type backgroundRouter struct {
-	gossipSub, err := pubsub.NewGossipSub(ctx, cfg.Host)
-	if err != nil {
-		return nil, fmt.Errorf("creating gossip pubsub: %w", err)
-	}
+	// TECHDEBT(#595): add ctx to interface methods and propagate down.
+	ctx, cancel := context.WithCancel(context.TODO())
 
-	dhtMode := dht.ModeAutoServer
-	// NB: don't act as a bootstrap node in peer discovery in client debug mode
-	if isClientDebugMode(bus) {
-		dhtMode = dht.ModeClient
+	rtr := &backgroundRouter{
+		logger:    networkLogger,
+		handler:   cfg.Handler,
+		host:      cfg.Host,
+		cancelCtx: cancel,
 	}
+	rtr.SetBus(bus)
 
-	kadDHT, err := dht.New(ctx, cfg.Host, dht.Mode(dhtMode))
-	if err != nil {
-		return nil, fmt.Errorf("creating DHT: %w", err)
+	if err := rtr.setupDependencies(ctx, cfg); err != nil {
+		return nil, err
 	}
 
-	topic, err := gossipSub.Join(protocol.BackgroundTopicStr)
-	if err != nil {
-		return nil, fmt.Errorf("joining background topic: %w", err)
-	}
+	go rtr.readSubscription(ctx)
 
-	// INVESTIGATE: `WithBufferSize` `SubOpt`:
-	// > WithBufferSize is a Subscribe option to customize the size of the subscribe
-	// > output buffer. The default length is 32 but it can be configured to avoid
-	// > dropping messages if the consumer is not reading fast enough.
-	// (see: https://pkg.go.dev/github.com/libp2p/go-libp2p-pubsub#WithBufferSize)
-	subscription, err := topic.Subscribe()
-	if err != nil {
-		return nil, fmt.Errorf("subscribing to background topic: %w", err)
-	}
+	return rtr, nil
+}
 
-	rtr := &backgroundRouter{
-		host:         cfg.Host,
-		gossipSub:    gossipSub,
-		kadDHT:       kadDHT,
-		topic:        topic,
-		subscription: subscription,
-		logger:       networkLogger,
-		pstore:       pstore,
+func (rtr *backgroundRouter) Close() error {
+	rtr.logger.Debug().Msg("closing background router")
+
+	rtr.cancelCtx()
+	rtr.subscription.Cancel()
+
+	var topicCloseErr error
+	if err := rtr.topic.Close(); err != context.Canceled {
+		topicCloseErr = err
 	}
 
-	return rtr, nil
+	return multierr.Append(
+		topicCloseErr,
+		rtr.kadDHT.Close(),
+	)
 }
 
 // Broadcast implements the respective `typesP2P.Router` interface  method.
-func (rtr *backgroundRouter) Broadcast(data []byte) error {
+func (rtr *backgroundRouter) Broadcast(pocketEnvelopeBz []byte) error {
+	backgroundMsg := &typesP2P.BackgroundMessage{
+		Data: pocketEnvelopeBz,
+	}
+	backgroundMsgBz, err := proto.Marshal(backgroundMsg)
+	if err != nil {
+		return err
+	}
+
 	// TECHDEBT(#595): add ctx to interface methods and propagate down.
-	return rtr.topic.Publish(context.TODO(), data)
+	return rtr.topic.Publish(context.TODO(), backgroundMsgBz)
 }
 
 // Send implements the respective `typesP2P.Router` interface  method.
-func (rtr *backgroundRouter) Send(data []byte, address cryptoPocket.Address) error {
+func (rtr *backgroundRouter) Send(pocketEnvelopeBz []byte, address cryptoPocket.Address) error {
+	rtr.logger.Warn().Str("address", address.String()).Msg("sending background message to peer")
```

Comment:
TL;DR I think state sync depends on unstaked actors supporting unicast send. I would demote this to a debug log.

---

I was actually considering converting this to a debug log (similar to [`rainTreeRouter#sendInternal()`](https://github.com/pokt-network/pocket/blob/main/p2p/raintree/router.go#L147)).

One case where we "need" to support unstaked actor to unstaked actor unicast is for the debug CLI. Granted, it's a weak case to make the argument; however, should lead to reduced complexity in the debug CLI (and possibly raintree's peer manager) as it no longer has to consider a scenario where it's not part of the staked actor set.

The stronger case to make the argument would be that, if my understanding is correct, state sync depends on unicast for responses. I'm under the impression that this is the case given [the current state](https://github.com/pokt-network/pocket/pull/812/files) of #812. Please correct me if I'm mistaken.

Again, correct me if I'm mistaken, but I think it may be the case that background router

Reply:
(relates to the tables in the [P2P Module / Router Decoupling](https://github.com/pokt-network/pocket/blob/bf96542fa6620d338b42184f7bad497f277eabe9/p2p/README.md#p2p-module--router-decoupling) section of the updated P2P README)